### DISCUSSION TITLE
What's the consistency model for documents?

### DISCUSSION THREAD

**Discussion by danielkeller** [n0-computer/iroh#2581] [General]
I can't seem to find this mentioned in the documentation. Can two users set the same key in the same document? What happens if they do this at the same time?

**Comment by matheus23** [comment]
Iroh documents are built on top of an older draft state of the willow protocol specification: https://willowprotocol.org/

The short answer is: Events have a timestamp (wall clock time by default) and the newer event wins.
But the internal state keeps the lastest state for each by node ID around in principle.
